server/sonar-server/src/main/java/org/sonar/server/authentication/OAuth2ContextFactory.java

```diff
@@ -26,28 +26,27 @@
 import org.sonar.api.server.authentication.OAuth2IdentityProvider;
 import org.sonar.api.server.authentication.UserIdentity;
 import org.sonar.api.utils.MessageException;
-import org.sonar.db.DbClient;
 import org.sonar.db.user.UserDto;
 import org.sonar.server.authentication.event.AuthenticationEvent;
-import org.sonar.server.user.ServerUserSession;
 import org.sonar.server.user.ThreadLocalUserSession;
+import org.sonar.server.user.UserSessionFactory;
 
 import static java.lang.String.format;
 import static org.sonar.api.CoreProperties.SERVER_BASE_URL;
 import static org.sonar.server.authentication.OAuth2CallbackFilter.CALLBACK_PATH;
 
 public class OAuth2ContextFactory {
 
-  private final DbClient dbClient;
   private final ThreadLocalUserSession threadLocalUserSession;
   private final UserIdentityAuthenticator userIdentityAuthenticator;
   private final Server server;
   private final OAuthCsrfVerifier csrfVerifier;
   private final JwtHttpHandler jwtHttpHandler;
+  private final UserSessionFactory userSessionFactory;
 
-  public OAuth2ContextFactory(DbClient dbClient, ThreadLocalUserSession threadLocalUserSession, UserIdentityAuthenticator userIdentityAuthenticator, Server server,
-    OAuthCsrfVerifier csrfVerifier, JwtHttpHandler jwtHttpHandler) {
-    this.dbClient = dbClient;
+  public OAuth2ContextFactory(ThreadLocalUserSession threadLocalUserSession, UserIdentityAuthenticator userIdentityAuthenticator, Server server,
+    OAuthCsrfVerifier csrfVerifier, JwtHttpHandler jwtHttpHandler, UserSessionFactory userSessionFactory) {
```

Comment:
usually, order of constructor arguments matches the order of properties and order they are assigned in the constructor

personnally, I tend to match the order of properties with the order they are used